app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp

```diff
@@ -1098,6 +1100,17 @@ DeviceDelegateOculusVR::SetRenderMode(const device::RenderMode aMode) {
   vrb::RenderContextPtr render = m.context.lock();
   for (int i = 0; i < VRAPI_EYE_COUNT; ++i) {
     m.eyeSwapChains[i]->Init(render, m.renderMode, m.renderWidth, m.renderHeight);
+    // auto surfaceOut = vrapi_GetTextureSwapChainAndroidSurface(m.eyeSwapChains[i]->ovrSwapChain);
+    // surfaceOut is null from vrapi_GetTextureSwapChainAndroidSurface(m.eyeSwapChains[i]->ovrSwapChain);.
+  }
+
+  for (int i = 0; i < DeviceDelegateOculusVR::State::EYE_SURFACE_COUNT; ++i) {
+    if (!m.eyeSurfaceSwapChain[i]) {
+      m.eyeSurfaceSwapChain[i] = vrapi_CreateAndroidSurfaceSwapChain(m.renderWidth, m.renderHeight);
```

Comment:
SwapChains are being recreated twice, first for standalone mode then for immersive. If they need to be created before immersive render mode is set, you can use GetImmersiveRenderSize() instead of m.renderWidth (they may have different sizes for immersive and standalone)